addons/hw_drivers/controllers/driver.py

```diff
@@ -40,8 +40,8 @@ def action(self, session_id, device_identifier, data):
                     _logger.info("Ignored request from %s as iot_idempotent_id %s already received from session %s",
                                  session_id, iot_idempotent_id, idempotent_session)
                     return False
-            iot_device.action(data)
-            return True
+            result = iot_device.action(data)
+            return result if result is not None else True
```

Comment:
this function `action(self, session_id, device_identifier, data)` aims to return the request action to the correct device. So what this function should return is "did you send to the correct device yes or no"
At no time should it return anything other than this response.
Here it returns the response from the driver function.

Have you looked at how others drivers return information in Odoo?